<?xml version='1.0' ?>
<launch>

  <arg name="use_sim_time" default="false" doc="Whether or not we are using simulation time steps"/>

  <arg name="fleet_name" doc="Name of the fleet."/>
  <arg name="robot_name" doc="Name or unique identifier of the robot."/>
  <arg name="robot_model" doc="Model or make of the robot."/>
  <arg name="level_name" doc="The name of the current level the robot is on."/>
  
  <arg name="dds_domain" default="42" doc="DDS domain ID for the CycloneDDS middleware."/>
  <arg name="node_name" default="$(arg robot_name)_client_node" doc="Node name of the client."/>
  <arg name="move_base_server" default="move_base" doc="Name of move_base server."/>
  <arg name="set_map_server" default="set_map" doc="Name of the map setting server."/>
  <arg name="get_map_servers" default="" doc="map_name: get_map_server_name"/>

  <arg name="battery_state_topic" default="battery_state" doc="Battery state topic."/>
  <arg name="map_frame" default="map" doc="Global map frame that the robot uses."/>
  <arg name="robot_frame" default="base_footprint" doc="Robot base frame that the robot uses for transforms."/>

  <arg name="frequency" default="10" doc="Frequency at which the client operates at."/>
  
  <node name="client_node" pkg="free_fleet_ros1" type="client" output="screen">
    <param name="fleet_name" type="string" value="$(arg fleet_name)"/>
    <param name="robot_name" type="string" value="$(arg robot_name)"/>
    <param name="robot_model" type="string" value="$(arg robot_model)"/>
    <param name="level_name" type="string" value="$(arg level_name)"/>
    <param name="dds_domain" type="int" value="$(arg dds_domain)"/>
    <param name="node_name" type="string" value="$(arg node_name)"/>
    <param name="move_base_server" type="string" value="$(arg move_base_server)"/>
    <param name="set_map_server" type="string" value="$(arg set_map_server)"/>
    <param name="battery_state_topic" type="string" value="$(arg battery_state_topic)"/>
    <param name="map_frame" type="string" value="$(arg map_frame)"/>
    <param name="robot_frame" type="string" value="$(arg robot_frame)"/>
    <param name="frequency" type="int" value="$(arg frequency)"/>
    <param name="use_sim_time" value="$(arg use_sim_time)"/>
  </node>

</launch>
